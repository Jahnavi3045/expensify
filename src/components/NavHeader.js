import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogOut } from '../actions/auth'


export const NavHeader = ({startLogOut}) => {
  return (
    <div>
        <h2>Expensify</h2>
        <NavLink to={'/'}>Home</NavLink>
        <NavLink to={'/addExpense'}>addExpense</NavLink>
        <NavLink to={'/help'}>Help</NavLink>
        <button onClick={startLogOut}>log out</button>
    </div>
  )
}

const mapDispatchToProps=(dispatch)=>({
  startLogOut:()=>dispatch(startLogOut())
})
export default connect(undefined,mapDispatchToProps)(NavHeader);
