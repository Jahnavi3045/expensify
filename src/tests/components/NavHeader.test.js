import React from "react";
import ReactShallowRenderer from 'react-test-renderer/shallow'
import NavHeader from '../../components/NavHeader'

test('should render header correctly',()=>{
    const renderer=new ReactShallowRenderer()
    renderer.render(<NavHeader/>)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
})