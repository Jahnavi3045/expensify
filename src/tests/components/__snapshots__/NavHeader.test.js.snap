// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<div>
  <h2>
    Expensify
  </h2>
  <NavLink
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    to="/addExpense"
  >
    addExpense
  </NavLink>
  <NavLink
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
